include/RAJA/policy/sycl/kernel/SyclKernel.hpp

```diff
@@ -48,19 +48,22 @@ namespace RAJA
  * work group and work items per group.
  */
 template <bool async0>
-struct sycl_launch {};
+struct sycl_launch : public RAJA::make_policy_pattern_launch_platform_t<
+                            RAJA::Policy::sycl,
+                            RAJA::Pattern::forall,
+                            detail::get_launch<async0>::value,
+                            RAJA::Platform::sycl>{
+};
 
 namespace statement
 {
 
-
-/*! RAJA::kernel statement that launches a SYCL kernel.
- *
- *
+/*
+ * ! RAJA::kernel statement that launches a SYCL kernel.
  */
 template <typename LaunchConfig, typename... EnclosedStmts>
 struct SyclKernelExt
-    : public internal::Statement<sycl_launch<0>, EnclosedStmts...> {
+    : public internal::Statement<LaunchConfig, EnclosedStmts...> {
```

Comment:
Is this consistent with the cuda/hip back-ends?

Reply:
Well, we don't have all the variations that CUDA and HIP have. The `LaunchConfig` template parameter was not used in the original version. I added the inheritance on the launch platform type above so we could get the correct resource type and async/sync execution behavior. I think it's pretty close, but not exact. If you have something else in mind, please let me know.

Reply:
Its fine, I just wanted to be sure we were thinking about keeping it relatively similar.